packages/lib/src/components/ThreeDS2/components/DelegatedAuthentication/DelegatedAuthenticationEnrollment.tsx

```diff
@@ -0,0 +1,260 @@
+        onComplete(data);
+    };
+
+    const completeWithErrorCode = (errorCode: number) => {
+        const sdkOutput = encodeDelegatedAuthenticationEnrollmentOutputData({ errorCode });
+        complete(sdkOutput);
+    };
+
+    const completeWithErrorDescription = (errorDescription: string) => {
+        const sdkOutput = encodeDelegatedAuthenticationEnrollmentOutputData({ errorDescription });
+        complete(sdkOutput);
+    };
+
+    useEffect(() => {
+        const checkSupported = async () => {
+            try {
+                if (!window.PublicKeyCredential) {
+                    completeWithErrorCode(DelegatedAuthenticationError.WebauthnNotSupported);
+                }
+
+                const uvpaAvailable = await window.PublicKeyCredential.isUserVerifyingPlatformAuthenticatorAvailable();
+                // we only want to perform browser DA on devices with a platform authenticator
+                if (!uvpaAvailable) {
+                    completeWithErrorCode(DelegatedAuthenticationError.NoPlatformAuthenticator);
+                }
+
+                const supportsSpc = await isSpcSupported();
+                if (inIframe && !supportsSpc) {
+                    // merchant has an iframe integration meaning we can only do spc,
+                    // but spc is not supported, so do not enroll credential
+                    completeWithErrorCode(DelegatedAuthenticationError.SPCNotSupported);
+                }
+
+                setSpcSupported(supportsSpc);
+                setSupported(true);
+            } catch (e) {
+                console.error(e);
+                completeWithErrorCode(DelegatedAuthenticationError.Unknown);
+            }
+        };
+
+        checkSupported();
+    }, []);
+
+    useEffect(() => {
+        if (!supported) {
+            return;
+        }
+
+        if (!token) {
+            completeWithErrorCode(DelegatedAuthenticationError.NoInputDataPresent);
+        }
+
+        try {
+            const parsedRegistrationData = parseEnrollmentData(token);
+            setRegistrationData(parsedRegistrationData);
+        } catch (e) {
+            completeWithErrorDescription(e.message);
+        }
+    }, [supported]);
+
+    useEffect(() => {
+        setStatus('start');
+    }, [registrationData]);
+
+    useEffect(() => {
+        if (!paymentCredential) {
+            return;
+        }
+
+        setCredentialCookie(paymentCredential.id);
+        submitPaymentCredential(paymentCredential);
+    }, [paymentCredential]);
+
+    const parseEnrollmentData = (enrollmentDataString: string) => {
+        const decodedEnrollmentDataString = new TextDecoder().decode(decodeBase64(enrollmentDataString));
+        return JSON.parse(decodedEnrollmentDataString);
+    };
+
+    const isSpcSupported = async () => {
+        if (!window.PaymentRequest) {
+            return false;
+        }
+
+        try {
+            return await mockPaymentRequest().canMakePayment();
```

Comment:
Does this mean everytime a user does DA a mock payment request is made first to determine if the payment can be made?